#! /bin/sh -e

# DP: Add Debian URL for bug reporting instructions.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir=$3/
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
	patch $pdir -f --no-backup-if-mismatch -p0 < $0
	if grep -q debian_bug_report_url ${dir}gcc/version.c; then
	  :
	else
	  echo 'const char debian_bug_report_url[] = "<URL:file:///usr/share/doc/gcc-4.1/README.Bugs>";' >> ${dir}gcc/version.c
	fi
	;;
    -unpatch)
	grep -v debian_bug_report_url ${dir}gcc/version.c > ${dir}gcc/version.c.new
	${dir}move-if-change ${dir}gcc/version.c.new ${dir}gcc/version.c
	patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
	;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- gcc/java/jcf-dump.c.orig	2005-09-05 00:22:28.368267856 +0000
+++ gcc/java/jcf-dump.c	2005-09-05 00:22:44.643793600 +0000
@@ -921,6 +921,8 @@
   printf ("\n");
   printf (_("For bug reporting instructions, please see:\n"
 	    "%s.\n"), bug_report_url);
+  printf (_("For Debian GNU/Linux specific bug reporting instructions, please see:\n"
+	    "%s.\n"), debian_bug_report_url);
   exit (0);
 }
 
--- gcc/diagnostic.c.orig	2005-09-05 00:22:28.371267400 +0000
+++ gcc/diagnostic.c	2005-09-05 00:22:44.644793448 +0000
@@ -243,6 +243,8 @@
       fnotice (stderr, "Please submit a full bug report,\n"
 	       "with preprocessed source if appropriate.\n"
 	       "See %s for instructions.\n", bug_report_url);
+      fnotice (stderr, "For Debian GNU/Linux specific bug reporting instructions,\n"
+	       "see %s.\n", debian_bug_report_url);
       exit (FATAL_EXIT_CODE);
 
     case DK_FATAL:
--- gcc/gcc.c.orig	2005-09-05 00:22:28.378266336 +0000
+++ gcc/gcc.c	2005-09-05 00:24:35.547933608 +0000
@@ -2872,9 +2872,11 @@
 	      fatal ("\
 Internal error: %s (program %s)\n\
 Please submit a full bug report.\n\
-See %s for instructions.",
+See %s for instructions.\n\
+For Debian GNU/Linux specific bug reporting instructions, see\n\
+%s.\n",
 		     strsignal (WTERMSIG (status)), commands[i].prog,
-		     bug_report_url);
+		     bug_report_url, debian_bug_report_url);
 	    signal_count++;
 	    ret_code = -1;
 	  }
@@ -6359,6 +6361,8 @@
 	{
 	  printf (_("\nFor bug reporting instructions, please see:\n"));
 	  printf ("%s.\n", bug_report_url);
+	  printf ("For Debian GNU/Linux specific bug reporting instructions, please see:\n");
+	  printf ("%s.\n", debian_bug_report_url);
 
 	  return (0);
 	}
@@ -6648,6 +6652,8 @@
     {
       printf (("\nFor bug reporting instructions, please see:\n"));
       printf ("%s\n", bug_report_url);
+      printf ("For Debian GNU/Linux specific bug reporting instructions, please see:\n");
+      printf ("%s\n", debian_bug_report_url);
     }
 
   return (signal_count != 0 ? 2
--- gcc/gccbug.in.orig	2005-09-05 00:22:28.381265880 +0000
+++ gcc/gccbug.in	2005-09-05 00:25:56.216670096 +0000
@@ -25,10 +25,10 @@
 VERSION=3.113
 
 # The submitter-id for your site.
-SUBMITTER=net
+SUBMITTER=net # net-debian
 
 # The default mail address for PR submissions. 
-GNATS_ADDR=gcc-gnats@gcc.gnu.org
+GNATS_ADDR="gcc-gnats@gcc.gnu.org,debian-gcc@lists.debian.org"
 
 # The default release for this host.
 # We have to guess at what program_transform_name might have done.
--- gcc/gcov.c.orig	2005-09-05 00:22:28.382265728 +0000
+++ gcc/gcov.c	2005-09-05 00:22:44.650792536 +0000
@@ -402,6 +402,8 @@
   fnotice (file, "  -u, --unconditional-branches    Show unconditional branch counts too\n");
   fnotice (file, "\nFor bug reporting instructions, please see:\n%s.\n",
 	   bug_report_url);
+  fnotice (file, "\nFor Debian GNU/Linux specific bug reporting instructions, please see:\n%s.\n",
+	   debian_bug_report_url);
   exit (status);
 }
 
--- gcc/version.h.orig	2005-09-05 00:22:28.383265576 +0000
+++ gcc/version.h	2005-09-05 00:22:44.651792384 +0000
@@ -2,4 +2,5 @@
 #define GCC_VERSION_H
 extern const char version_string[];
 extern const char bug_report_url[];
+extern const char debian_bug_report_url[];
 #endif /* ! GCC_VERSION_H */
